### ISSUE TITLE
Highly non-trivial contraction crash

### ISSUE THREAD

**Issue by doctorn** [homotopy-io/homotopy-rs#814]
I have no idea how to minimise this.

[homotopy_io_crash_action_dump(1).txt](https://github.com/homotopy-io/homotopy-rs/files/9568233/homotopy_io_crash_action_dump.1.txt)

**Comment by NickHu** [comment]
Possibly fixed by a02898f9629e12544f916c4d8c094b4b1fbb095a

I ran your action log through homotopy-cli and it didn't crash at least:

```
Performing action: CreateGeneratorZero
Performing action: SelectGenerator(0:0⁺)
Performing action: TakeIdentityDiagram
Performing action: SetBoundary(Source)
Performing action: SelectGenerator(0:0⁺)
Performing action: TakeIdentityDiagram
Performing action: SetBoundary(Target)
Performing action: EditSignature(Edit(Node(2), MakeInvertible(1:2⁺, true)))
Performing action: SelectGenerator(1:2⁺)
Performing action: SetBoundary(Source)
Performing action: SelectGenerator(0:0⁺)
Performing action: TakeIdentityDiagram
Performing action: TakeIdentityDiagram
Performing action: SelectPoints([[Boundary(Target), Boundary(Source)], [Boundary(Target), Interior(Regular(0))], [Interior(Regular(0)), Interior(Regular(0))]])
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: Attach(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] })
Performing action: SetBoundary(Target)
Performing action: SelectGenerator(1:2⁺)
Performing action: SelectPoints([[Boundary(Target), Interior(Regular(0))]])
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: Attach(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] })
Performing action: SelectPoints([[Boundary(Target), Boundary(Source)], [Boundary(Target), Interior(Regular(0))], [Interior(Regular(2)), Interior(Regular(0))]])
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: HighlightAttachment(None)
Performing action: HighlightAttachment(None)
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: None, boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: None, boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: Attach(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁺, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: None, boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] })
Performing action: SelectPoints([[Boundary(Target), Boundary(Target)], [Boundary(Target), Interior(Regular(0))], [Interior(Regular(3)), Interior(Regular(0))]])
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: HighlightAttachment(Some(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] }))
Performing action: Attach(AttachOption { generator: 1:2⁺, diagram: DiagramN { source: DiagramN { source: Diagram0(0:0⁺), cospans: [] }, cospans: [Cospan { forward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 0), [Regular(0)]))))], singular_slices: [] } }] }, backward: RewriteN { dimension: 1, cones: [Cone { index: 0, internal: ConeInternal { source: [], target: Cospan { forward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 1), [])))), backward: Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Target, 1), [])))) }, regular_slices: [Rewrite0(0:0⁺, 1:2⁻, Label(Some((1, BoundaryPath(Source, 0), [Regular(0)]))))], singular_slices: [] } }] } }] }, tag: Some("inverse"), boundary_path: Some(BoundaryPath(Target, 0)), embedding: [0] })
Performing action: TakeIdentityDiagram
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 2, direction: Backward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 1, direction: Backward }))
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 2, direction: Backward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 1, direction: Backward }))
Performing action: AscendSlice(1)
Performing action: TakeIdentityDiagram
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Forward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: SwitchSlice(Backward)
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [Interior(Regular(2))], height: 2, direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 3, direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 3, direction: Backward }))
Performing action: Homotopy(Expand(Expand { location: [Interior(Singular(2)), Interior(Singular(0))], direction: Backward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 2, direction: Backward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 0, direction: Forward }))
Performing action: AscendSlice(1)
Performing action: UpdateView(View { dimension: 3 })
Performing action: UpdateView(View { dimension: 2 })
Performing action: HighlightSlice(Some(Boundary(Target)))
Performing action: HighlightSlice(Some(Boundary(Target)))
Performing action: DescendSlice(Boundary(Target))
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(None)
Performing action: AscendSlice(1)
Performing action: HighlightSlice(Some(Interior(Regular(3))))
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(Some(Boundary(Target)))
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(Some(Boundary(Target)))
Performing action: HighlightSlice(Some(Boundary(Target)))
Performing action: DescendSlice(Boundary(Target))
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(None)
Performing action: HighlightSlice(None)
Performing action: Homotopy(Contract(Contract { bias: Some(Lower), location: [Interior(Regular(2))], height: 2, direction: Forward }))
Performing action: Homotopy(Expand(Expand { location: [Interior(Singular(2)), Interior(Singular(2)), Interior(Singular(0))], direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 1, direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 3, direction: Backward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Lower), location: [Interior(Regular(2))], height: 1, direction: Forward }))
Performing action: Homotopy(Expand(Expand { location: [Interior(Singular(2)), Interior(Singular(1)), Interior(Singular(0))], direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 3, direction: Forward }))
Performing action: Homotopy(Expand(Expand { location: [Interior(Singular(3)), Interior(Singular(1)), Interior(Singular(0))], direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 2, direction: Backward }))
Performing action: Homotopy(Expand(Expand { location: [Interior(Singular(1)), Interior(Singular(1)), Interior(Singular(0))], direction: Forward }))
Performing action: SelectPoints([[Interior(Regular(2)), Interior(Singular(2))]])
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [Interior(Regular(3))], height: 1, direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 4, direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 3, direction: Backward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 1, direction: Forward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 3, direction: Backward }))
Performing action: Homotopy(Contract(Contract { bias: Some(Same), location: [], height: 1, direction: Forward }))
Performing action: SelectPoints([[Interior(Singular(1)), Interior(Singular(0))]])
```

**Comment by NickHu** [comment]
Imported in the tool and it contracts without issue
